posts/2024-09-21-seal_of_approval-collapse/index.qmd

```diff
@@ -0,0 +1,128 @@
+---
+title: "Seal of Approval: collapse"
+author: "Sebastian Krantz"
+date: "Sept 21, 2024"
+categories: [seal of approval, partner package]
+image: "hex.png"
+draft: true
+---
+
+## [`collapse`](https://github.com/SebKrantz/collapse)
+
+*Author(s):*	Sebastian Krantz
+
+*Maintainer:*	Sebastian Krantz (sebastian.krantz@graduateinstitute.ch)
+
+![collapse hex sticker](hex.png)
+
+`collapse` is a large C/C++-based infrastructure package facilitating complex statistical computing, data transformation, and exploration tasks in R - at outstanding levels of performance and memory efficiency. It also implements a class-agnostic approach to R programming supporting vector, matrix and data frame-like objects (includings *xts*, *tibble*, ***data.table***, and *sf*). It has a stable API, depends on *Rcpp*, and supports R versions >= 3.4.0. 
+
+## Relationship with `data.table`
+
+At the C-level, `collapse` took much inspiration from `data.table`, and leverages some of its core algorithms like radixsort, while adding significant [statistical functionality](https://sebkrantz.github.io/collapse/reference/collapse-documentation.html) and [new algorithms](https://sebkrantz.github.io/collapse/reference/fast-grouping-ordering.html) within a [class-agnostic programming framework](https://sebkrantz.github.io/collapse/articles/collapse_object_handling.html) that seamslessly supports `data.table`. Notably, [`collapse::qDT()`](https://sebkrantz.github.io/collapse/reference/quick-conversion.html) is a highly efficient anything to `data.table` converter, and all manipulation functions in `collapse`, if passed `data.table`'s, return valid `data.table`'s, allowing for subsequent reference operations (`:=`). 
```

Comment:
> ... that seamslessly supports `data.table`.

seamlessly

Reply:
> and all manipulation functions in `collapse`, if passed `data.table`'s, return valid `data.table`'s, allowing for subsequent reference operations (`:=`).

This could do with slight rephrasing

Perhaps 'and all manipulation functions in `collapse` return a valid `data.table` object when a `data.table` is passed, enabling subsequent reference operations (:=).'